flytekit/core/type_engine.py

```diff
@@ -437,7 +447,29 @@ def to_literal_type(cls, python_type: Type) -> LiteralType:
         Converts a python type into a flyte specific ``LiteralType``
         """
         transformer = cls.get_transformer(python_type)
-        return transformer.get_literal_type(python_type)
+        res = transformer.get_literal_type(python_type)
+        data = None
+        if hasattr(python_type, "__metadata__"):
```

Comment:
instead, let's use [get_args](https://docs.python.org/3/library/typing.html#typing.get_args) to list the annotations.

Reply:
sure but why

Reply:
just to bullet-proof this code. dunder methods/attributes are not guaranteed to stay consistent between versions, as per https://docs.python.org/3/reference/lexical_analysis.html#reserved-classes-of-identifiers.

Reply:
ok that makes sense